core/models/user.py

```diff
@@ -0,0 +1,54 @@
+from django.contrib.auth.base_user import BaseUserManager
+from django.contrib.auth.models import AbstractUser
+from django.utils.translation import gettext_lazy as _
+
+from django.db import models
+
+
+class UserManager(BaseUserManager):
+    """Define a model manager for User model with email field."""
+
+    use_in_migrations = True
+
+    def _create_user(self, email, password, **extra_fields):
+        """Create and save a User with the given email and password."""
+        if not email:
+            raise ValueError('The given email must be set')
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
```

Comment:
Need to check: is it hashed in database after saving?

Reply:
I think it is hashed. https://docs.djangoproject.com/en/4.0/ref/contrib/auth/#django.contrib.auth.models.User.check_password